File: python/Unidad_3/ej1.py
# Abra una imagen y agréguele ruido con el comando imnoise

import matplotlib.pyplot as plt
from skimage.io import imread
from skimage.util import random_noise as noise
import tkinter as tk
from tkinter import filedialog


def file_read():
    file_path = filedialog.askopenfilename()
    return file_path


root = tk.Tk()
root.withdraw()
try:
    # Busco imagen y obtengo su ruta
    image = imread(file_read())
    img_sp = noise(image, mode='s&p', seed=200)
    img_gs = noise(image, mode='gaussian', seed=200)

    fig, axes = plt.subplots(ncols=3, nrows=1, figsize=(10, 2.5))
    ax = axes.ravel()
    ax[0] = plt.subplot(131)
    ax[1] = plt.subplot(132)
    ax[2] = plt.subplot(133)

    ax[0].imshow(image)
    ax[0].set_title('Original')

    ax[1].imshow(img_sp)
    ax[1].set_title('Salt & Pepper')

    ax[2].imshow(img_gs)
    ax[2].set_title('Gaussian')

except:
    print('Cerraste la ventana!')

plt.show()
print('Adios!')
root.destroy()

File: matlab/Unidad_3/ej1.m
% Abra una imagen y agréguele ruido con el comando imnoise.

[file, path] = uigetfile({'*.tiff';'*.JPG';'*.bmp';'*.jpg'});
dir = strcat(path,file);
info = imfinfo(dir);
I=imread(dir);

switch info.ColorType
    case 'grayscale'
        I=imread(dir);
    case 'truecolor'
        rgb=imread(dir);
        I=rgb2gray(rgb);
    case 'indexed'
        [X,map]=imread(dir);
        I=ind2gray(X,map);
end

% Ruidos
IN1 = imnoise(I, 'salt & pepper', 0.05);
IN2 = imnoise(I, 'gaussian', 0, 0.01); 
IN3 = imnoise(I, 'speckle', 0.04);

% Plots
subplot(2,2,1); imshow(I);
title('Imagen original');

subplot(2,2,2); imshow(IN1)
title('Salt & Pepper');

subplot(2,2,3); imshow(IN2)
title('Gaussian');

subplot(2,2,4); imshow(IN3)
title('Speckle');